app/channel_category.go

```diff
@@ -144,14 +144,135 @@ func (a *App) UpdateSidebarCategoryOrder(userId, teamId string, categoryOrder []
 }
 
 func (a *App) UpdateSidebarCategories(userId, teamId string, categories []*model.SidebarCategoryWithChannels) ([]*model.SidebarCategoryWithChannels, *model.AppError) {
-	result, err := a.Srv().Store.Channel().UpdateSidebarCategories(userId, teamId, categories)
+	updatedCategories, originalCategories, err := a.Srv().Store.Channel().UpdateSidebarCategories(userId, teamId, categories)
 	if err != nil {
 		return nil, model.NewAppError("UpdateSidebarCategories", "app.channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
 	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_SIDEBAR_CATEGORY_UPDATED, teamId, "", userId, nil)
 	a.Publish(message)
-	return result, nil
+
+	a.muteChannelsForUpdatedCategories(userId, updatedCategories, originalCategories)
+
+	return updatedCategories, nil
+}
+
+func (a *App) muteChannelsForUpdatedCategories(userId string, updatedCategories []*model.SidebarCategoryWithChannels, originalCategories []*model.SidebarCategoryWithChannels) {
+	var channelsToMute []string
+	var channelsToUnmute []string
+
+	// Mute or unmute all channels in categories that were muted or unmuted
+	for i, updatedCategory := range updatedCategories {
+		originalCategory := originalCategories[i]
```

Comment:
I'll prefer a length check somewhere that verifies length of both slices are same, rather than just directly accessing it.

Reply:
Yep, I can add that for safety. They should be the same length, but "should" can be dangerous